### PR TITLE
Make Kedro not instantiate datasets from kedro.extras.datasets

### LINKED ISSUE #2125: Make Kedro not instantiate datasets from `kedro.extras.datasets`
### Description
From `0.19.0` Kedro templates/starters will just use datasets from `kedro-datasets` directly and not refer to `kedro.extras.datasets` anymore

### PR DESCRIPTION
**Pull request by AhdraMeraliQB**
> **NOTE:** Kedro datasets are moving from `kedro.extras.datasets` to a separate `kedro-datasets` package in
> [`kedro-plugins` repository](https://github.com/kedro-org/kedro-plugins). Any changes to the dataset implementations
> should be done by opening a pull request in that repository.
## Description

Resolves #2125 
Starters imports already resolved by https://github.com/kedro-org/kedro-starters/pull/109

## Checklist

- [x] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [x] Signed off each commit with a [Developer Certificate of Origin (DCO)](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/managing-repository-settings/managing-the-commit-signoff-policy-for-your-repository)
- [x] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [x] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [ ] Added tests to cover my changes
- [ ] Checked if this change will affect Kedro-Viz, and if so, communicated that with the Viz team

### CODE REVIEW

**Review by noklam — COMMENTED** [review]
I expect there are tests to be updated, either in test/io/core or test/data_catalog.py (not sure where). Can you check with this?

**Comment by AhdraMeraliQB** [comment]
@noklam 

> I expect there are tests to be updated, either in test/io/core or test/data_catalog.py (not sure where). Can you check with this?

I couldn't find any other references to `kedro.extras...` or `kedro[...DataSet]` in the codebase; `test/io/test_data_cataog.py` and `tests/framework/cli/test_catalog.py` already import from `kedro_datasets` as per #3057

**Comment by AhdraMeraliQB** [comment]
@noklam 

> I expect there are tests to be updated, either in test/io/core or test/data_catalog.py (not sure where). Can you check with this?

I couldn't find any other references to `kedro.extras...` or `kedro[...DataSet]` in the codebase; `test/io/test_data_cataog.py` and `tests/framework/cli/test_catalog.py` already import from `kedro_datasets` as per #3057

File: docs/source/nodes_and_pipelines/nodes.md [hunk 1]
```diff
@@ -295,7 +295,7 @@ import pandas as pd
 from kedro.io.core import (
     get_filepath_str,
 )
-from kedro.extras.datasets.pandas import CSVDataSet
+from kedro_datasets.pandas import CSVDataSet
```
[the remark this answers is not in the document]
Comment (AhdraMeraliQB) [reply docs/source/nodes_and_pipelines/nodes.md:298]: Would the tests still work before we release the changes on `kedro-datasets`?

**Comment by noklam** [comment]
@AhdraMeraliQB You are right! I was looking at the `main` branch accidentally, looks good to me then!

**Review by noklam — APPROVED** [review]

File: docs/source/nodes_and_pipelines/nodes.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (merelcht) [reply docs/source/nodes_and_pipelines/nodes.md:298]: Ahh good point.. let's leave it for now and just do it as part of the final rename.

**Review by merelcht — APPROVED** [review]

**Pull request merged**